### PR TITLE
Update institution discover default query options

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: []
-   Feature flag: n/a

## Purpose
- Actual fetch stuff in the institution discover page
- Only fetch stuff from the specific environment the search is happening in by using `accessService` query-param
  - This addresses the issue of seeing multiple instances of `OSF Registries` or `Open-ended Registration` registration templates on staging

## Summary of Changes
- Add `rorIri` and `iri` attributes to Institution model (ultimately these comes from the `ror_uri ` and `identifier_domain` respectively in the admin app and is being added in a [BE PR](https://github.com/CenterForOpenScience/osf.io/pull/10452))
- fetch by either ROR, URI, or URL from OSFv2 URL (the URLs that are [relevant to SHARE for understanding institution affiliation](https://github.com/CenterForOpenScience/osf.io/blob/a3e0a0b9ddda5dd75fc8248d58f3bcdeece0323e/osf/metadata/osf_gathering.py#L811-L816))
- Fetch either by `affilation`, `creator,affiliation` (for preprints), `isContainedBy,affiliation` (for files)
- Add `accessService` qp to SHARE requests if we are not serving locally

### CODE REVIEW

**Review by aaxelb — COMMENTED** [review]
this'll need backend changes to work as desired, sorry to say

File: app/institutions/discover/controller.ts [hunk 1]
```diff
@@ -17,8 +16,11 @@ export default class InstitutionDiscoverController extends Controller {
     queryParams = ['cardSearchText', 'sort', 'resourceType'];
 
     get defaultQueryOptions() {
+        const identifiers = [this.model.rorIri, this.model.iri, this.model.links.self].filter(Boolean).join(',');
         return {
-            affiliation: pathJoin(config.OSF.url, 'institutions', this.model.id),
+            affiliation: identifiers,
+            'creator,affiliation': identifiers,
+            'isContainedby,affiliation': identifiers,
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply app/institutions/discover/controller.ts:23]: Ahh, good catch, I had completely blanked that adding filters like this wasn't going to work! I agree with BrianG and also am in favor of using the comma-separated list of period-delimited paths. Will make those changes now.

**Review by aaxelb — CHANGES_REQUESTED** [review]
looks good, tho i think it'll need a few more changes to work with the change to `.`-delimited paths in query params

specifically: `RelatedPropertyPathModel.shortFormLabel` still joins on `,` -- may be best to remove that `shortFormLabel` getter and switch to `propertyPathKey` in the few places it's still used

**Review by aaxelb — APPROVED** [review]
🥦

**Pull request merged**